tasks/git/src/main/java/com/walmartlabs/concord/plugins/git/tokens/AccessTokenProvider.java

```diff
@@ -0,0 +1,51 @@
+package com.walmartlabs.concord.plugins.git.tokens;
+
+/*-
+ * *****
+ * Concord
+ * -----
+ * Copyright (C) 2017 - 2025 Walmart Inc., Concord Authors
+ * -----
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =====
+ */
+
+import com.walmartlabs.concord.plugins.git.GitSecretService;
+import com.walmartlabs.concord.plugins.git.Utils;
+import com.walmartlabs.concord.plugins.git.model.Auth;
+
+import java.nio.file.Files;
+
+public interface AccessTokenProvider {
+
+    String getToken();
+
+    static AccessTokenProvider fromAuth(Auth auth, String baseUrl, String installationRepo, GitSecretService secretService) {
+
+        if (auth instanceof Auth.AccessTokenAuth tokenAuth) {
```

Comment:
I think with sealed interfaces we could use `switch` with pattern matching here?

```java
switch (auth) {
  case Auth.AccessToken -> ...
  case Auth.AppInstallationAuth -> ...
  ...etc
}
```

Reply:
Ah, right. We should move to 21 sometime soon:)